source/module_io/bessel_basis.cpp

```diff
@@ -194,13 +195,20 @@ void Bessel_Basis::init_TableOne(
     ModuleBase::GlobalFunc::OUT(GlobalV::ofs_running, "dr",dr);
 
 	// allocate rmesh and Jlk and eigenvalue of Jlq
-	double *r = new double[rmesh];
-	double *rab = new double[rmesh];
-	double *jle = new double[rmesh];
-	double *jlk = new double[rmesh];
-	double *g = new double[rmesh]; // smooth function
-	double *function = new double[rmesh];
-	double *en = new double[ecut_number];
+	// double *r = new double[rmesh];
+	// double *rab = new double[rmesh];
+	// double *jle = new double[rmesh];
+	// double *jlk = new double[rmesh];
+	// double *g = new double[rmesh]; // smooth function
+	// double *function = new double[rmesh];
+	// double *en = new double[ecut_number];
+    std::vector<double> r(rmesh);
+    std::vector<double> rab(rmesh);
+    std::vector<double> jle(rmesh);
+    std::vector<double> jlk(rmesh);
+    std::vector<double> g(rmesh);
+    std::vector<double> function(rmesh);
+    std::vector<double> en(rmesh);
```

Comment:
the dimension is ecut_number?

Reply:
Sorry, teacher. I've changed the dimension to ecut_number now.